### PR TITLE
refactor(ui): flatten `ui/src/app` dir 

### PR DESCRIPTION
**Pull request by agilgur5**
Fixes #12539 "Future Work" part 2

### Motivation

Remove unnecessary nesting, same as #12539

### Modifications

- move `ui/src/models` to `ui/src/shared/models`
  - fix-up various imports to account for this
- then flatten `ui/src/app` to just `ui/src/`

- also fix name of `workflow-operation-map.tsx` to `.ts` as it doesn't have JSX/TSX (fixes https://github.com/argoproj/argo-workflows/pull/11121#discussion_r1561987261)

- also move `webpack.config.js` out of `ui/src/app` and to `ui/`
  - same with `tsconfig.json`
  - slightly modify directories in the config files to account for this
  - remove symlinks and config that was previously necessary due to having config in the subdirectory

### Verification

1. `make start UI=true` loads the UI fine and dandy and it works as before
1. `yarn lint` and `yarn build` etc all pass
1. `make server/static/files.go STATIC_FILES=true` works/passes
1. `ls dist/app` looks about the same before/after:

<details>

before:
```
❯ ls dist/app      
775.43e8c969ca85676f528d.js                             argo-ui-logs-viewer.ae9615e8f7f5b591df8e.js.map         monaco-editor.fa9e75b9e342fbb964bb.js.LICENSE.txt
775.43e8c969ca85676f528d.js.LICENSE.txt                 assets                                                  monaco-editor.fa9e75b9e342fbb964bb.js.map
775.43e8c969ca85676f528d.js.map                         codicon.ttf                                             react-markdown-plus-gfm.3a77fefac0c77fa0b385.js
839.58e22fba88631d094daf.js                             editor.worker.js                                        react-markdown-plus-gfm.3a77fefac0c77fa0b385.js.map
839.58e22fba88631d094daf.js.LICENSE.txt                 editor.worker.js.map                                    react-monaco-editor.9c2ee7a658ee5378c65b.js
839.58e22fba88631d094daf.js.map                         index.html                                              react-monaco-editor.9c2ee7a658ee5378c65b.js.map
990.e6d98ce60f5195aec70a.js                             json.worker.js                                          reports.17861f7c283f72c6ea9b.js
990.e6d98ce60f5195aec70a.js.LICENSE.txt                 json.worker.js.LICENSE.txt                              reports.17861f7c283f72c6ea9b.js.map
990.e6d98ce60f5195aec70a.js.map                         json.worker.js.map                                      swagger-ui-react-css.dfee358b653e751976b0.js
994.eae31bc7956069a7bb1d.js                             main.e0032a966b1e70172630.js                            swagger-ui-react-css.dfee358b653e751976b0.js.map
994.eae31bc7956069a7bb1d.js.LICENSE.txt                 main.e0032a966b1e70172630.js.LICENSE.txt                swagger-ui-react.e89a642a6a2e1d7b1970.js
994.eae31bc7956069a7bb1d.js.map                         main.e0032a966b1e70172630.js.map                        swagger-ui-react.e89a642a6a2e1d7b1970.js.LICENSE.txt
argo-ui-logs-viewer.ae9615e8f7f5b591df8e.js             monaco-editor.fa9e75b9e342fbb964bb.js                   swagger-ui-react.e89a642a6a2e1d7b1970.js.map
```

after:
```
❯ ls dist/app/
775.43e8c969ca85676f528d.js                           argo-ui-logs-viewer.ae9615e8f7f5b591df8e.js.map       monaco-editor.92ae73621991273cd7fa.js.LICENSE.txt
775.43e8c969ca85676f528d.js.LICENSE.txt               assets/                                               monaco-editor.92ae73621991273cd7fa.js.map
775.43e8c969ca85676f528d.js.map                       codicon.ttf                                           react-markdown-plus-gfm.d507f040b86c97f5d729.js
839.58e22fba88631d094daf.js                           editor.worker.js                                      react-markdown-plus-gfm.d507f040b86c97f5d729.js.map
839.58e22fba88631d094daf.js.LICENSE.txt               editor.worker.js.map                                  react-monaco-editor.9c2ee7a658ee5378c65b.js
839.58e22fba88631d094daf.js.map                       index.html                                            react-monaco-editor.9c2ee7a658ee5378c65b.js.map
990.e6d98ce60f5195aec70a.js                           json.worker.js                                        reports.aeeaa52ae94633cc90fc.js
990.e6d98ce60f5195aec70a.js.LICENSE.txt               json.worker.js.LICENSE.txt                            reports.aeeaa52ae94633cc90fc.js.map
990.e6d98ce60f5195aec70a.js.map                       json.worker.js.map                                    swagger-ui-react-css.dfee358b653e751976b0.js
994.eae31bc7956069a7bb1d.js                           main.4c6024a7ded227c8d4f0.js                          swagger-ui-react-css.dfee358b653e751976b0.js.map
994.eae31bc7956069a7bb1d.js.LICENSE.txt               main.4c6024a7ded227c8d4f0.js.LICENSE.txt              swagger-ui-react.e89a642a6a2e1d7b1970.js
994.eae31bc7956069a7bb1d.js.map                       main.4c6024a7ded227c8d4f0.js.map                      swagger-ui-react.e89a642a6a2e1d7b1970.js.LICENSE.txt
argo-ui-logs-viewer.ae9615e8f7f5b591df8e.js           monaco-editor.92ae73621991273cd7fa.js                 swagger-ui-react.e89a642a6a2e1d7b1970.js.map
```

</details>

### Notes to Reviewers

- Use the sidebar in GH to see the files that weren't just renamed, which is only a small handful and mostly the last few config files

- This will merge conflict with some of my other open PRs that delete or change imports; can update depending on what is merged first

### CODE REVIEW

File: ui/src/models/event-source.ts
Comment (agilgur5) [inline ui/src/models/event-source.ts:1]: the `{}` -> `any` change made this a full delete/add instead of a rename

File: ui/src/app/tsconfig.json
Comment (agilgur5) [inline ui/src/app/tsconfig.json:1]: not sure why `git` didn't detect this as a rename 🤔

File: ui/tsconfig.json
Comment (agilgur5) [inline ui/tsconfig.json:1]: `outDir` here and `output.path` in `webpack.config.js` need a bit of carefulness since they're used in the `Dockerfile`

**Review by tczhao — APPROVED** [review]
Note: `tsconfig.json` and `webpack.config.js` are moved to their default path so that we don't need to specify the location

**Pull request merged**